exp/api/v1alpha4/azuremanagedmachinepool_webhook_test.go

```diff
@@ -94,6 +94,45 @@ func TestAzureManagedMachinePoolUpdatingWebhook(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "Cannot add AvailabilityZones after creating agentpool",
+			new: &AzureManagedMachinePool{
+				Spec: AzureManagedMachinePoolSpec{
+					Mode:         "System",
+					SKU:          "StandardD2S_V3",
+					OSDiskSizeGB: to.Int32Ptr(512),
+				},
+			},
+			old: &AzureManagedMachinePool{
+				Spec: AzureManagedMachinePoolSpec{
+					Mode:              "System",
+					SKU:               "StandardD2S_V3",
+					OSDiskSizeGB:      to.Int32Ptr(512),
+					AvailabilityZones: []string{"1", "2", "3"},
```

Comment:
shouldn't it be the opposite? old has no zones, new has zones

Reply:
@CecileRobertMichon you are correct. Sorry for the delay. 

I have updated the code according to the review comments